packages/camera/camera_windows/lib/camera_windows.dart

```diff
@@ -0,0 +1,437 @@
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math';
+
+import 'package:camera_platform_interface/camera_platform_interface.dart';
+import 'package:camera_windows/src/utils/utils.dart';
+import 'package:cross_file/cross_file.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:stream_transform/stream_transform.dart';
+
+/// An implementation of [CameraPlatform] for Windows.
+class CameraWindows extends CameraPlatform {
+  /// Registers the Windows implementation of CameraPlatform.
+  static void registerWith() {
+    CameraPlatform.instance = CameraWindows();
+  }
+
+  /// The method channel used to interact with the native platform.
+  final MethodChannel _pluginChannel =
+      const MethodChannel('plugins.flutter.io/camera');
+
+  /// Camera specific method channels to allow comminicating with specific cameras.
+  final Map<int, MethodChannel> _cameraChannels = <int, MethodChannel>{};
+
+  /// The controller that broadcasts events coming from handleCameraMethodCall
+  ///
+  /// It is a `broadcast` because multiple controllers will connect to
+  /// different stream views of this Controller.
+  /// This is only exposed for test purposes. It shouldn't be used by clients of
+  /// the plugin as it may break or change at any time.
+  @visibleForTesting
+  final StreamController<CameraEvent> cameraEventStreamController =
+      StreamController<CameraEvent>.broadcast();
+
+  /// Returns a stream of camera events for the given [cameraId].
+  Stream<CameraEvent> _cameraEvents(int cameraId) =>
+      cameraEventStreamController.stream
+          .where((CameraEvent event) => event.cameraId == cameraId);
+
+  @override
+  Future<List<CameraDescription>> availableCameras() async {
+    try {
+      final List<Map<dynamic, dynamic>>? cameras = await _pluginChannel
+          .invokeListMethod<Map<dynamic, dynamic>>('availableCameras');
+
+      if (cameras == null) {
+        return <CameraDescription>[];
+      }
+
+      return cameras.map((Map<dynamic, dynamic> camera) {
+        return CameraDescription(
+          name: camera['name'] as String,
+          lensDirection:
+              parseCameraLensDirection(camera['lensFacing'] as String),
+          sensorOrientation: camera['sensorOrientation'] as int,
+        );
+      }).toList();
+    } on PlatformException catch (e) {
+      throw CameraException(e.code, e.message);
+    }
+  }
+
+  @override
+  Future<int> createCamera(
+    CameraDescription cameraDescription,
+    ResolutionPreset? resolutionPreset, {
+    bool enableAudio = false,
+  }) async {
+    try {
+      final Map<String, dynamic>? reply = await _pluginChannel
+          .invokeMapMethod<String, dynamic>('create', <String, dynamic>{
+        'cameraName': cameraDescription.name,
+        'resolutionPreset': resolutionPreset != null
+            ? _serializeResolutionPreset(resolutionPreset)
+            : null,
```

Comment:
What does it mean to have a `null` resolution preset? It doesn't look like it's documented on the interface. I assume the camera default?

We should document this in the parent interface.

Reply:
Currently plugin uses best available resolution if resolutionPreset value is null. Current plugin implementation loops cameras resolutions (media_types) through and selects best resolution fitting to the resolutionPreset. If resolution preset is null, best available resolution is selected. Using cameras default media type is one option, but would need separate implementation.
In most cases users use CameraController from main camera plugin which requires resolutionPreset value as non-nullable. But this of course should be documented somewhere.

Reply:
Not sure should this value ever be nullable for any platform as it seems to throw error on both Android and iOS if null value is given.